arch/Xtensa/XtensaMapping.c

```diff
@@ -0,0 +1,206 @@
+#include <capstone/xtensa.h>
+
+#include "../../MCRegisterInfo.h"
+#include "../../MCInst.h"
+#include "../../SStream.h"
+#include "../../Mapping.h"
+#include "../../utils.h"
+#include "../../cs_simple_types.h"
+#include "XtensaDisassembler.h"
+#include "XtensaInstPrinter.h"
+#include "priv.h"
+
+#ifndef CAPSTONE_DIET
+
+static const char *const insn_name_maps[] = {
+#include "XtensaGenCSMappingInsnName.inc"
+};
+
+static const name_map group_name_maps[] = {
+#include "XtensaGenCSFeatureName.inc"
+};
+
+const insn_map mapping_insns[] = {
+#include "XtensaGenCSMappingInsn.inc"
+};
+
+static const map_insn_ops insn_operands[] = {
+#include "XtensaGenCSMappingInsnOp.inc"
+};
+
+#endif
+
+#define GET_REGINFO_MC_DESC
+#include "XtensaGenRegisterInfo.inc"
+
+void Xtensa_init_mri(MCRegisterInfo *mri)
+{
+	MCRegisterInfo_InitMCRegisterInfo(
+		mri, XtensaRegDesc, ARR_SIZE(XtensaRegDesc), 0, 0,
+		XtensaMCRegisterClasses, ARR_SIZE(XtensaMCRegisterClasses), 0,
+		0, XtensaRegDiffLists, NULL, XtensaSubRegIdxLists,
+		ARR_SIZE(XtensaSubRegIdxLists), XtensaRegEncodingTable);
+}
+
+void Xtensa_printer(MCInst *MI, SStream *OS, void *info)
+{
+	Xtensa_LLVM_printInstruction(MI, MI->address, OS);
+}
+
+static void set_instr_map_data(MCInst *MI)
+{
+#ifndef CAPSTONE_DIET
+	map_cs_id(MI, mapping_insns, ARR_SIZE(mapping_insns));
+	map_implicit_reads(MI, mapping_insns);
+	map_implicit_writes(MI, mapping_insns);
+	map_groups(MI, mapping_insns);
+
+	unsigned opcode = MCInst_getOpcode(MI);
+	if (opcode > ARR_SIZE(insn_operands)) {
+		return;
+	}
+
+	const map_insn_ops *ops = &insn_operands[opcode];
+	cs_xtensa *detail = &MI->flat_insn->detail->xtensa;
```

Comment:
```suggestion
	cs_xtensa *detail = Xtensa_get_detail(MI);
```
Requires of course to add it in `Mapping.h`